### ISSUE TITLE
Handling network disconnection

### ISSUE THREAD

**Issue by ravidreams** [tshrinivasan/OCR4wikisource#21]
If there is network disconnection while do_OCR.py is running, it gets stuck in the same step. Only if i intervene by CTRL+C after network connection is live, it goes to next page. 

For example, if it gets stuck at 19th page, after CTRL+C it goes to 20th page. After the whole process, I find text file for page 19 is empty. So, we have to manually note all these problem pages and then manually do OCR later and upload.

**Comment by jayantanth** [comment]
++ Ravi

**Comment by jayantanth** [comment]
Proposal..........

After using this script last few days , I am very much agreed with Ravi,s thought. I am requesting to Shrini to add more feature during do_ocr.py.
1. This not for connection issue. Some we need to stop/pause/cancel.
   2.I have watched that , If I  stuck at 19th page, press after CTRL+C, the page 20 will start for uploading and page 19.pdf file also move to temp folder. I think it should not moved . 

Because if I run  the script next time , the remaining file should upload for ocr.
1. So my proposal , when I run next time  do_OCR.py it should not download again/split again. It should upload remaining pdf files.

**Comment by tshrinivasan** [comment]
Working on this.

Will update in few days.

**Comment by jayantanth** [comment]
Also I have watched that .... when I have skipped ( by pressing  CTRL+C ) in Three (3) times in my last action of 753 page book. I have found 750 pages OCRed text. ( may be because of skipped by me 3 times)  Randomly checked that text file content  and page content just wrong order. (  for exmple "text_for_page_00150.txt" should "text_for_page_00151.txt" OR "text_for_page_00750.txt" should be "text_for_page_00753.txt)

If I run  mediawiki_uploader.py, that will hit the wrong page number of wikisource. So I have to run again do_ocr.

[do_ocr_2016-01-07-23-23-24_log.txt](https://github.com/tshrinivasan/OCR4wikisource/files/83218/do_ocr_2016-01-07-23-23-24_log.txt)

**Comment by jayantanth** [comment]
So If I skipped page no 19 , then "text_for_page_00019.txt" should not be exist at folder.

**Comment by tshrinivasan** [comment]
Fixed this in 1.35

Now, it wont download the files if already exists.
No djvu to pdf conversion , if already done ins same folder.

Disconnections on google drive upload process can now resume for remaining files only.

Test this and share the results.

**Comment by tshrinivasan** [comment]
Fixed the skipping uploads in version 1.38

No of individual pages should be equal to the no of relevant text files.

If we skip manually or automatically on the upload process, it wont proceed further.

We have to rerun the script to upload the pending files.

It will process the text files only after all the PDF files are uploaded and received their text content.

Check this and share the results.

**Comment by jayantanth** [comment]
I think this issues was solved by closing #29  So would request to @tshrinivasan  please closed this issue.
